libvirt/tests/src/virtual_network/qemu/netperf.py

```
@@ -0,0 +1,863 @@
+        use_vmxml = params.get('use_vmxml', 'yes') == 'yes'
+        backup_vmxml = None
+        if use_vmxml:
+            vm = env.get_vm(vm_name)
+            if vm is None:
+                env.create_vm(vm_type='libvirt', target=None, name=vm_name, params=params, bindir=test.bindir)
+                vm = env.get_vm(vm_name)
+            
+            backup_vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+            vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+
+            vm_attrs = eval(params.get('vm_attrs', '{}'))
```

Ruff (0.14.10)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)